packages/auth/src/authClient.ts

```diff
@@ -380,34 +383,77 @@ export class AuthClient {
 
   /**
    * Initiates passkey (WebAuthn) authentication flow.
+   *
    * This method uses WebAuthn standards to authenticate users with biometric or hardware security keys.
+   * Can be used for both new passkey creation (when credentialId is undefined) and
+   * authentication with existing passkeys (when credentialId is provided).
    *
+   * @param {string} [credentialId] - Optional credential ID for authenticating with existing passkey
    * @returns {Promise<AuthSession>} Promise that resolves to an auth session instance
-   * @throws {Error} Currently throws "Not implemented" as this method is under development
    *
    * @example
    * ```ts
+   * // New passkey authentication
    * const authSession = await authClient.loginWithPasskey();
+   *
+   * // Authenticate with existing passkey
+   * const authSession = await authClient.loginWithPasskey("existing-credential-id");
    * ```
    */
-  public async loginWithPasskey(): Promise<AuthSession> {
+  public async loginWithPasskey(credentialId?: string): Promise<AuthSession> {
     // TODO: figure out what the current passkey code is doing.
+    // For new passkey authentication, credentialId would be undefined initially
     const stamper = await this.createWebAuthnStamper({
-      credentialId: undefined,
+      credentialId: credentialId ?? undefined,
```

Comment:
nit: the type here is already `string | undefined`, so you can simplify this:
```suggestion
      credentialId,
```